packages/modules/create-imodel/src/components/imodel-name/IModelName.tsx

```diff
@@ -0,0 +1,50 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import { LabeledInput } from "@itwin/itwinui-react";
+import React from "react";
+
+import { IModelContext } from "../context/imodel-context";
+
+export type IModelNameProps = {
+  /** iModel name property. */
+  nameString?: string;
+  /** Error message when name is too long. */
+  nameTooLong?: string;
+  /** Message to be displayed below input field */
+  message?: string;
+  /** Maximum length */
+  maxLength?: number;
+};
+
+export function IModelName(props: IModelNameProps) {
+  const MAX_LENGTH = props.maxLength || 255;
+
+  const isPropertyInvalid = (value: string) => {
```

Comment:
this is being used in other components too, so better move it to some utils file and reuse

Reply:
moved to utils